url80587fc7d5_dennervidal_CloudSim_DVFS_tgz-pJ8-Examples_Dvfs_example_simpleJ8/src/Examples/TLM_dvfs.java

```diff
@@ -0,0 +1,668 @@
+	 * @param userId
+	 * @param nb_cloudlet : number of cloudlet we want to create
+	 * @param IdShift : id of cloudlet
+	 * @param type_cl : type of cloudlet related to the XML file
+	 *
+	 * @return the cloudlet list
+	 */
+	private static List<Cloudlet> createCloudletList(int userId, int nb_cloudlet, int IdShift, int type_cl) {
+		List<Cloudlet> list = new ArrayList<Cloudlet>();
+
+		long length;
+		int pesNumber;
+		long fileSize;
+		long outputSize;
+		int offset = no_cur_cloudlet;
+	//	System.out.println("IdShift = " + IdShift );
+		
+		int CloudletID;
+
+		for (int i = no_cur_cloudlet ; i < (offset+nb_cloudlet) ; i++) 
+		{
+				CloudletDatas tmp_cl = vect_cls.get(type_cl);
+
+				length = tmp_cl.getLength();
+				pesNumber = tmp_cl.getPes();
+				fileSize = tmp_cl.getFilesize();
+				outputSize = tmp_cl.getOutputsize();
+
+				CloudletID=IdShift+list.size()+(type_cl*NbProcessusTLM);	
+
+				Cloudlet cloudlet = new Cloudlet(CloudletID, length, pesNumber, fileSize, outputSize, new UtilizationModelFull(), new UtilizationModelFull(), new UtilizationModelFull());
+				cloudlet.setUserId(userId);
+				cloudlet.setVmId(CloudletID); 
+				list.add(cloudlet);
+				no_cur_cloudlet++;
+			//	System.out.println("Cloudlet created // No de Cloudlet =  "+ CloudletID +"  //  Cloudlet List Size = " + list.size());
+		}
+		return list;
+	}
+
+	public static void UpdateCloudletList(List<Cloudlet> list_)
+	{
+		cloudletList.addAll(list_);
+	}
+
+	/**
+	 * Creates the vms.
+	 *
+	 * @param userId
+	 * @param nb_vm : number of cloudlet we want to create
+	 * @param IdShift : id of cloudlet
+	 * @param type_vm : type of cloudlet related to the XML file
+	 * @return the list< vm>
+	 */
+
+	private static List<Vm> createVms(int userId, int nb_vm, int IdShift, int type_vm) {
+		List<Vm> vms = new ArrayList<Vm>();
+
+		VmDatas tmp_vm = vect_vms.get(type_vm);
+
+
+		// VM description
+		int mips ;
+		int pesNumber;
+		int ram ;
+		long bw;
+		long size ;
+		String vmm;
+
+		System.out.println(no_cur_vm + "//" + nb_vm);
+		int offset=no_cur_vm;
+
+		int VmID;
+		for (int i = no_cur_vm ; i < (offset+nb_vm); i++) {
+				mips = tmp_vm.getMips();
+				pesNumber = tmp_vm.getCpu();
+				ram=tmp_vm.getRam();
+				bw=tmp_vm.getBw();
+				size=tmp_vm.getSize();
+				vmm=tmp_vm.getVmm();
+
+				VmID=IdShift+vms.size()+(type_vm*NbProcessusTLM);
+
+				vms.add(new Vm(VmID, userId, mips, pesNumber, ram, bw, size, vmm,  new CloudletSchedulerSpaceShared()));
+				no_cur_vm++;
+				//System.out.println("VM created  // No VM =  "+ VmID +"  //  List Vm size = " + vms.size() + "  // ID Shift+1 = " + (IdShift+i) + "id vm = " + vms.get(0).getId());
+		}
+
+		return vms;
+	}
+
+	public static void UpdateVmList(List<Vm> list_)
+	{
+		vmList.addAll(list_);
+	}
+
+
+
+	/**
+	 * Creates the datacenter.
+	 *
+	 * @param name the name
+	 *
+	 * @return the datacenter
+	 *
+	 * @throws Exception the exception
+	 */
+	private static PowerDatacenter createDatacenter(String name) throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 76 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 429-455 into a separate method.